### PR TITLE
Disable float typespec of StElem, when it is off

### PR DESCRIPTION
**Pull request by meg-gupta**

### CODE REVIEW

**Comment by meg-gupta** [comment]
Ping

File: lib/Backend/GlobOpt.cpp [hunk 1]
```diff
@@ -13477,8 +13477,11 @@ GlobOpt::TypeSpecializeStElem(IR::Instr ** pInstr, Value *src1Val, Value **pDstV
     case ObjectType::Float64VirtualArray:
     case ObjectType::Float64MixedArray:
     Float64Array:
-        toType = TyFloat64;
-        break;
+    if (this->DoFloatTypeSpec())
```
Comment (LouisLaf) [inline lib/Backend/GlobOpt.cpp]: We also need to disable all of the int cases if DoAggressiveIntTypeSpec or DoFloatTypeSpec is disabled, like I did in ValueNumberLdElemDst()

File: lib/Backend/GlobOpt.cpp [hunk 2]
```diff
@@ -13450,7 +13450,10 @@ GlobOpt::TypeSpecializeStElem(IR::Instr ** pInstr, Value *src1Val, Value **pDstV
...
     case ObjectType::Int32MixedArray:
     Int32Array:
-        toType = TyInt32;
+        if (this->DoAggressiveIntTypeSpec() || this->DoFloatTypeSpec())
+        {
+            toType = TyInt32;
+        }
         break;
 
     case ObjectType::Uint32Array:
```
Comment (meg-gupta) [inline lib/Backend/GlobOpt.cpp]: @LouisLaf When the sym is already int/float specialized, I have not added the check to see if float/int typespec was disabled. This should be okay ?

**Comment by LouisLaf** [comment]
:shipit:

**Pull request merged**